src/bp/ui-studio/src/web/components/Layout/StatusBar/ConfigStatus.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, Classes, Colors, Dialog, Icon, Intent } from '@blueprintjs/core'
+import axios from 'axios'
+import React, { Fragment, useEffect, useState } from 'react'
+import { toastFailure } from '~/components/Shared/Utils'
+import EventBus from '~/util/EventBus'
+
+import ActionItem from './ActionItem'
+import style from './StatusBar.styl'
+
+const adminUrl = `${window['ROOT_PATH']}${window['API_PATH']}/admin/server/`
+
+const ConfigStatus = () => {
+  const [isDifferent, setDifferent] = useState(false)
+  const [isRestarting, setRestart] = useState(false)
+  const [isOpen, setOpen] = useState(false)
+
+  useEffect(() => {
+    // tslint:disable-next-line: no-floating-promises
+    fetchHash()
+
+    const configUpdated = event => setDifferent(event.initialHash !== event.newHash)
+
+    EventBus.default.on('config.updated', configUpdated)
+    return () => EventBus.default.off('config.updated', configUpdated)
+  }, [])
+
+  useEffect(() => {
+    if (!isRestarting) {
+      return
+    }
+
+    const interval = setInterval(async () => {
+      try {
```

Comment:
since you don't have anyting in your catch bloc you could use .then instead to make it more compact

```js
axios.get('/status', {timeout:500}).then(window.location.reload)
```